### ISSUE TITLE
Reduce potential for accidental disclosure of personal information

### ISSUE THREAD

**Issue by kimberscott** [lookit/lookit-api#321]
**TL;DR**: Make it easy to download data with potentially identifying information obscured, and scary to download things you should be careful with.

**Narrative**
As an Experimenter, I want to establish a reproducible data analysis pipeline and share the results of my research with others. However, some of the data I download may contain personal information about participants that needs to be removed before I can share it, and some files may not be appropriate to share at all. I want to get as little personal information as possible in the "default" files available for me to download. I also want to be reminded to be careful when downloading files that require special care.

**Preliminary decisions to make**
We could alternately provide age in days at time of participation but (by default) not include any timestamps. This would support even infancy studies and prevent researchers from having to combine child data with session data to get exact ages. But we would need to remove timestamps of sessions, demographic data, maybe consent rulings from the file downloads, and we'd need to make sure timestamps were removed from all event data in ember-lookit-frameplayer (e.g. by measuring from the start of the experiment). 

**Implementation Notes**
It will be important to be very clear to researchers about how age rounding works, and we probably want the rounding results to be independent of when in the year the child was born (i.e., we do not want to go by calendar months). To the nearest 10 or 30 days would be reasonable (or 10 until 360 and thereafter 30).
@kimberscott can write the warning text.

**TODO/Acceptance**

- [x] Add checkboxes on all responses page for including/not including identifying info in all responses (CSV only). Two columns: checkboxes for how to get age (birthdate, age in days, rounded age) and for what other data to include (either other child data, grouped, and other account data; or break down into gender, gest age, conditions, languages, addl info, name). 
- [x] Implement rounded age field/property/fn
- [x] Implement use of checkboxes (including for file naming) for CSV
- [x] Extend to JSON all response data
- [x] Set up individual response data to create JSON, overview CSV, or frame CSV
- [x] Add checkboxes for identifying info on individual response page & implement (CSV, JSON; disabled for frame CSV)
- [x] Add demographic/account data download button & implement. Remove list of birthdays and replace with list of approximate ages of children in home.
- [x] Add warning text to demographic data
- [x] Add warning/explanatory text for all response view
- [x] Participant view: remove/process list of birthdays as in account data download (separate, just for readability)
- [x] Add demographic data dictionary download option
- [x] Add child data download button & implement
- [x] Add child data dictionary download button & implement
- [x] Implement object uuid + study uuid hash for child, participant, and demographic snapshot (not response) for #127 (this way researchers working on the same study get the same hashed IDs to use regardless of who's downloading data, but a researcher who runs studies on different protocols doesn't get hashed IDs that link people across those studies)
   - [x] Figure out hash procedure / initial setup. 
   - [x] Set up demo for child ID
   - [x] Extend to participant and demographic snapshot ID
- [x] Add UUID/id (of child, account - not response) as fields that can be (by default) omitted from "all responses" file downloads
- [x] Add UUID as fields that can be omitted from child data download
- [x] Add UUID (of child, account - not response) as fields that can be (by default) omitted from "individual response" file download
- [x] Add UUID as field that can be omitted from demographic data download (or just omit)
- [x] Edit all data-dict text to explain difference between ID types
- [x] Call UUID "global ID" everywhere displayed to researcher
- [x] Use hashed study-specific participant and child IDs in place of "ID" in individual response display, consent manager display
- [x] Display download options on all responses & demo snapshots regardless of whether there are any responses, just disable download buttons if not
- [x] Decide whether/how to use hashed participant IDs in email view & implement
- [x] Create and use short readable hash of response ID as well - decide on implementation; add to individual responses, all responses, demographic data downloads as appropriate. Decided to stick with UUIDs because of use in video filenames, utility in looking up via API or checking for errors, etc, but use first segment for display in individual responses.
- [x] Edit Terms of Service to reflect that this is implemented (specifically say don't publish the global_id fields, and that hashed_id fields are fine)
- [x] Edit documentation and tutorial to explain these options (include estimate of when collisions would occur, why/how the salt)

Note that birthdays and nicknames will remain accessible via the API and on the "view participant" and consent pages to facilitate communication with families. Our goal is to provide a **default**, easy workflow that does not generate files with a bunch of names and birthdays - we do in fact trust researchers with them in principle.

**Comment by kimberscott** [comment]
May want to include #127 in some form
